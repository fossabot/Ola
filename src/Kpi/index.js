import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const Kpi = ({ title, value, description, status, className, ...props }) => {
  const styles = cx(
    'ola_kpi',
    { 'ola-skeleton': status !== 'loaded' },
    { 'is-loading': status === 'loading' },
    className
  )

  return (
    <div className={cx(styles)} {...props}>
      { status === 'loaded'
        ? <>
          <div className="ola_kpi-title">{title}</div> 
          <div className="ola_kpi-value">{value}</div>
          { description && <div className="ola_kpi-description">{description}</div>}
        </>
        : <>
          <div className="ola_kpi-title"></div> 
          <div className="ola_kpi-value"></div>
          <div className="ola_kpi-description"></div>
        </>
      }
    </div>
  )
}
  
Kpi.defaultProps = {
  title: null,
  value: null,
  description: null,
  status: 'loaded',
  className: null
}

Kpi.propTypes = {
  /** Kpi title */
  title: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
  /** Kpi Value */
  value: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Kpi Description */
  description: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Kpi status */
  status: PT.oneOf(['loaded', 'loading', 'empty']),
  /** Extra className */
  className: PT.string,
}

export default Kpi