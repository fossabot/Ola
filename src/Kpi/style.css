.ola_kpi {

    --skeleton-height: 50px;

    --title-height: var(--size-6);
    --title-width: var(--size-11);
    --title-position: 10px 10px;
    --title-skeleton: linear-gradient(var(--gray-xlight) var(--title-height), transparent 0);

    --value-height: var(--size-8);
    --value-width: var(--size-10);
    --value-position: 10px 40px;
    --value-skeleton: linear-gradient(var(--gray-xlight) var(--value-height), transparent 0);

    --description-height: var(--size-4);
    --description-width: var(--size-10);
    --description-position: 10px 90px;
    --description-skeleton: linear-gradient(var(--gray-xlight) var(--description-height), transparent 0);

    --skeleton-animation: linear-gradient(
        90deg, 
        rgba(255,255,255, 0) 0,
        rgba(255,255,255,.8) 50%,
        rgba(255,255,255,0) 100%,
        transparent 0
    );
    --blur-width: 200px;
    --blur-size: var(--blur-width) --skeleton-height;

    display: inline-block;

    &.is-positive {
        & .ola_kpi-description {
            color: var(--success);
        }
    }

    &.is-negative {
        & .ola_kpi-description {
            color: var(--error);
        }
    }

    &:empty {
        content:"";
        display:block;
        width: 100%;
        min-height: 150px;
        background-image: 
            var(--title-skeleton), 
            var(--value-skeleton),
            var(--description-skeleton);
        background-size: 
            var(--title-width) var(--title-height), 
            var(--value-width) var(--value-height), 
            var(--description-width) var(--description-height), 
            100% 100%;
        background-position: 
            var(--title-position), 
            var(--value-position),
            var(--description-position), 
            0 0;
        background-repeat: no-repeat;
    }

}

.ola_kpi-title {
    font: var(--font-body);
    font-weight: var(--regular);
    color: var(--black);
    margin: 0 0;
}

.ola_kpi-value {
    display: block;
    font: var(--font-title);
    color: var(--black);
    margin: var(--size-3) 0 0;
}

.ola_kpi-description {
    font: var(--font-caption);
    font-weight: var(--bold);
    color: var(--gray);
    margin: var(--size-1) 0 0;
}