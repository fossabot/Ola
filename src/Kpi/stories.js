import React from 'react'
import Kpi from './'

export default {
  title: 'Kpi',
  component: Kpi,
  args: {
    title: 'Estimated monthly visits comming from search engines',
    value: '166 visits',
    description: '55% more than in May',
    variant: 'positive'
  }
}

export const Base = (args) => <Kpi {...args} />

export const WithTag = () => 
    <div className="ola_ly-columns">
        <div>
            <Kpi 
              value={<p style={{margin: 0}}><img src="http://lorempixel.com/300/100/"/></p> }
              title={<p style={{margin: 0}}>Estimated <strong>monthly</strong> visits comming from search engines</p>} 
              description={<p style={{margin: 0}}>Estimated monthly visits comming from search engines</p>}
              variant="negative" />

        </div>
        <div>
            <Kpi 
              value="Value Test"
              title={<p style={{margin: 0}}>Estimated <strong>monthly</strong> visits comming from search engines</p>} 
              description={<p style={{margin: 0}}>Estimated monthly visits comming from search engines</p>}
              variant="positive" />
        </div>
    </div>