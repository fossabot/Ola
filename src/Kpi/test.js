import React from 'react'
import Kpi from './'
import renderer from 'react-test-renderer'

it('Default Kpi', () => {
  const tree = renderer
    .create(<Kpi 
      title='Estimated monthly visits comming from search engines'
      value='166 visits'
      description='55% more than in May'
      variant='positive'
      loading={false}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Kpi Loading', () => {
  const tree = renderer
    .create(<Kpi 
      title='Estimated monthly visits comming from search engines'
      value='166 visits'
      description='55% more than in May'
      variant='positive'
      loading={true}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})