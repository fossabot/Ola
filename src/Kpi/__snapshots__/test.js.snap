// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Kpi 1`] = `
<div
  className="ola_kpi is-positive"
>
  <div
    className="ola_kpi-title"
  >
    Estimated monthly visits comming from search engines
  </div>
  <div
    className="ola_kpi-value"
  >
    166 visits
  </div>
  <div
    className="ola_kpi-description"
  >
    55% more than in May
  </div>
</div>
`;

exports[`Kpi Loading 1`] = `
<div
  className="ola_kpi is-positive is-loading"
/>
`;
